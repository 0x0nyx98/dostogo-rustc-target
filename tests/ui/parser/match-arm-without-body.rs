macro_rules! pat {
    () => { Some(_) }
}

fn main() {
    match Some(false) {
        Some(_)
    }
    match Some(false) {
        Some(_)
        _ => {}
        //~^ ERROR expected one of
    }
    match Some(false) {
        Some(_),
        //~^ ERROR unexpected `,` in pattern
        //~| HELP try adding parentheses to match on a tuple
        //~| HELP or a vertical bar to match on multiple alternatives
    }
    match Some(false) {
        Some(_),
        //~^ ERROR unexpected `,` in pattern
        //~| HELP try adding parentheses to match on a tuple
        //~| HELP or a vertical bar to match on multiple alternatives
        _ => {}
    }
    match Some(false) {
        Some(_) if true
    }
    match Some(false) {
        Some(_) if true
        _ => {}
        //~^ ERROR expected one of
    }
    match Some(false) {
        Some(_) if true,
    }
    match Some(false) {
        Some(_) if true,
        _ => {}
    }
    match Some(false) {
        pat!()
    }
    match Some(false) {
        pat!(),
    }
    match Some(false) {
        pat!() if true,
    }
    match Some(false) {
        pat!()
        //~^ ERROR expected `,` following `match` arm
        //~| HELP missing a comma here
        _ => {}
    }
    match Some(false) {
        pat!(),
        _ => {}
    }
}
