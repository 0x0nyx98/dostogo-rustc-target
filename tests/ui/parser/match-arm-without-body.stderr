error: expected one of `,`, `=>`, `if`, `|`, or `}`, found reserved identifier `_`
  --> $DIR/match-arm-without-body.rs:11:9
   |
LL |         Some(_)
   |                - expected one of `,`, `=>`, `if`, `|`, or `}`
LL |         _ => {}
   |         ^ unexpected token

error: unexpected `,` in pattern
  --> $DIR/match-arm-without-body.rs:15:16
   |
LL |         Some(_),
   |                ^
   |
help: try adding parentheses to match on a tuple...
   |
LL |         (Some(_),)
   |         +        +
help: ...or a vertical bar to match on multiple alternatives
   |
LL |         Some(_) |
   |

error: unexpected `,` in pattern
  --> $DIR/match-arm-without-body.rs:21:16
   |
LL |         Some(_),
   |                ^
   |
help: try adding parentheses to match on a tuple...
   |
LL ~         (Some(_),
LL |
LL |
LL |
LL ~         _) => {}
   |
help: ...or a vertical bar to match on multiple alternatives
   |
LL ~         Some(_) |
LL +
LL +
LL +
LL ~         _ => {}
   |

error: expected one of `,`, `.`, `=>`, `?`, `}`, or an operator, found reserved identifier `_`
  --> $DIR/match-arm-without-body.rs:32:9
   |
LL |         Some(_) if true
   |                        - expected one of `,`, `.`, `=>`, `?`, `}`, or an operator
LL |         _ => {}
   |         ^ unexpected token

error: expected `,` following `match` arm
  --> $DIR/match-arm-without-body.rs:52:15
   |
LL |         pat!()
   |               ^ help: missing a comma here to end this `match` arm: `,`

error: aborting due to 5 previous errors

